src/hooks/usePagination.ts

```diff
@@ -0,0 +1,63 @@
+import { useCallback, useState } from 'react';
+
+type Props = {
+  page: number;
+  count: number;
+  onChange: (page: number) => void;
+};
+
+const usePagination = ({ page = 1, count = 1, onChange: handleChange }: Props) => {
```

Comment:
usePagination hook doesn't update properly the value to the outer world. Either changes to the dependency arrays or a useEffect to trigger the onChange function.

Reply:
nicely done catching this, just fixed with a cleaner solution by tracking the value change and trigger handleChange there